File: AceptaElReto/114-UltimoDigitoFactorial.java
import java.util.Scanner;

public class UltimoDigitoFactorial {

	public static void main(String[] args) {
		
		 Scanner s = new Scanner(System.in);
		 
		 int rep = s.nextInt();
		 int nums = 0;

		 while (rep>nums) {
			 
			int num = s.nextInt();
			System.out.println(lastDigitFactorial(num));
				 
			 nums++; 
		 }
	}
	 
	 public static int lastDigitFactorial(int n) {
		 
		 	switch (n) {
         		case 0:
         		case 1:	return 1;
         		case 2:	return 2;
         		case 3:	return 6;
         		case 4:	return 4;
         		default: return 0;
		 }
			
	 }
	
}

File: AceptaElReto/114-UltimoDigitoFactorial.c
#include <stdio.h>

int main() {
    
    int rep, num;
    scanf("%d", &rep);
    int nums = 0;

    while (rep > nums) {
        
        scanf("%d", &num);
        
        switch (num) {
            case 0:
            case 1:	
                num = 1;
                break;
            case 2:	
                num = 2;
                break;
            case 3:	
                num = 6;
                break;
            case 4:	
                num = 4;
                break;
            default: 
                num = 0;
        }

        printf("%d\n", num);
            
        nums++; 
    }

    return 0;
}